roles/pulpito/tasks/setup_pulpito.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: Checkout the repo
+  git:
+    repo: "{{ pulpito_repo }}"
+    dest: "{{ pulpito_repo_path }}"
+  become_user: "{{ pulpito_user }}"
+  tags:
+    - repos
+
+- name: Look for the virtualenv
+  stat: 
+    path: "{{ pulpito_repo_path }}/virtualenv"
+    get_checksum: no
+    get_md5: no
+  register: virtualenv
+
+- name: Create the virtualenv
+  shell: virtualenv ./virtualenv chdir={{ pulpito_repo_path }}
+  become_user: "{{ pulpito_user }}"
+  when: virtualenv.stat.exists == false
+
+- name: Self-upgrade pip
```

Comment:
You're about upgrading pip... if you do not do this on system with older pip it can't install all pulpito dependencies because required packages are working with newer pip. For example, openSUSE 42.3 have pip 7